diffrax/_step_size_controller/adaptive.py

```diff
@@ -179,6 +179,52 @@ class PIDController(
         common to refer to solving an equation to specific tolerances, without
         necessarily stating which solver was used.)
 
+        !!! Example
+
+            The choice of `rtol` and `atol` can have a significant impact on the
+            accuracy of even simple systems.
+            Consider a simple pendulum with a small angle kick:
+            ```python
+            class SimplePendulum(eqx.Module):
+                def __call__(self, t, state, args=None):
+                    theta, omega = state
+                    dtheta = omega
+                    domega = - jnp.sin(theta)
+                    return jnp.array([dtheta, domega])
+            dynamics = SimplePendulum()
+            init_angle = 0.1
+            init_omega = 0
+            state0 = jnp.array([init_angle, init_omega])
+            ```
+            We can integrate this using:
+            ```python
+            def integrator(dynamics, state0, stepsize_controller):
```

Comment:
I think it'd be good to make this example a little more concise. Perhaps something like this:
```python
import diffrax as dfx

def dynamics(t, y, args):
    dtheta = y["omega"]
    domega = - jnp.sin(y["theta"])
    return dict(theta=dtheta, omega=domega)

y0 = dict(theta=0.1, omega=0)
term = dfx.ODETerm(dynamics)
sol = dfx.diffeqsolve(
    term, solver, t0=0, t1=1000, dt0=0.1, y0,
    saveat=dfx.SaveAts(ts=jnp.linspace(0, 1000, 10000),
    max_steps=2**20,
    stepsize_controller=...
)
```
WDYT?

(WIth a little `...`, as we discuss different choices of that below.)

Reply:
That works well, 
yeah the class was a vestige of the much larger system I simplified for this, but this is much better as an example